app/lib/services/funding_eligibility.rb

```diff
@@ -63,12 +63,24 @@ def funding_eligiblity_status_code
       end
```

Comment:
This may be a solution that satisfies the competing desires of only querying the API once during all calculations and maintaining the services as having a single responsibility.

```
class FundingCalculator
  def initialize(...)
    @funding_calculation_data = FundingCalculationData.new(...)
  end

  def funding_eligibility
    FundingEligibilityCalculator.new(@funding_calculation_data).call
  end

  def targeted_delivery_funding_eligibility
    TargetedDeliveryFundingEligibilityCalculator.new(@funding_calculation_data).call
  end
end

class FundingCalculationData
  ...
    
  def api_call
    @api_call ||= ...
  end
end
```

This would move all the logic related to each type of eligibility into a single class with each being given a data store containing everything they need which would allow us to memoize the api call response.

Reply:
Overall however, whether anything needs changing or refactoring in this class is outside the scope of this PR and we'd need to have a large discussion about working in time to do this sort of refactor along with deciding what the correct approach is. 

Moving this into the model directory may be the simplest solution since this is clearly a model and not a service.